[Woo POS][Design System] Consolidate icon style

iamgabrielma opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #15113

## Description

Edit: I've updated the PR description with new images to not create confusion.

This PR consolidates the icon sizes into 3 `POSFontStyle`: small, medium, and large based on 1qcjzXitBHU7xPnpCOWnNM-fi-87_21502 


I have not modified the header back button icon/chevron, as I think makes more sense to tackle along the reusable header as a whole.

## Changes

| What                                       | Design Component  | Size   | File |
|--------------------------------------------|------------------|--------|--------------------------------------------------------------------|
| Remove item from card (x) button           | Product container | Medium | ItemRowView.swift |
| Exit POS (x) dismiss                       | Notice           | Small  | PointOfSaleExitPosAlertView.swift |
| Item image placeholder                     | Thumbnail        | Large  | POSItemImageView.swift |
| Exit modal (x) dismiss                     | Notice           | Small  | POSModalCloseButton.swift |
| Simple products header banner info circle  | Notice           | Large  | ItemListView.swift |
| Simple products header banner (x) dismiss  | Notice           | Small  | ItemListView.swift |
| Simple products header banner (i) toggle   | Page Header      | Large  | ItemListView.swift |


I've originally modified as well `PointOfSaleUnsupportedWidthView` and `POSErrorExclamationMark`, however I reverted these back to their 48/64 px original images, as we do not have sizing for those extra bigger images, neither a clear answer yet from the thread started here: p1739283631509359-slack-C070SJRA8DP


## Screenshots
* Item Row

<img width="299" alt="20250212_itemrow" src="https://github.com/user-attachments/assets/58a3f9ce-c7cd-4b3b-aad6-7ed382b49ab0" />

* Header banner expanded

<img width="546" alt="20250212_header_banner" src="https://github.com/user-attachments/assets/0385213c-96ee-4fe4-bf9d-a8000ffad481" />

* Header banner collapsed

<img width="593" alt="20250212_header_banner_collapsed" src="https://github.com/user-attachments/assets/355d5b74-f5fe-4bc0-83c5-6833df11cb93" />

## Testing information
- Enter in POS
- Observe the (i) icon in the header.
- Add some products to the cart, observe the (x) buttons
- Use split view and reduce the POS until is non eligible due to its size. Observe the non-eligible screen icon.
- Checkout with disconnected reader, observe the (!) reader not connected icon. Attempt to Connect and observe the (x) to dismiss the modal.
- Re-run the app by returning true from `ItemListView.shouldShowHeaderBanner`. Observe that both (¡) alert and dismiss (x) icons make sense.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

iamgabrielma committed:
Add preview

iamgabrielma committed:
Rename and apply posButtonSymbolSmall

dangermattic commented:
<!--
  0 Errors
  1 Warning: View files have been modified,...
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

iamgabrielma committed:
add posButtonSymbolLarge. apply to product limitation banner

iamgabrielma committed:
update PointOfSaleUnsupportedWidthView

iamgabrielma committed:
Update size for POSErrorExclamationMark

iamgabrielma committed:
Add posButtonSymbolMedium

staskus commented:
> ItemRowView's remove button is updated from 24 to 32. I think this one is very noticeable, maybe we need an extrasmall for these?

Yes, the size is very noticeable. A component in the designs uses much smaller icon:

<img width="404" alt="image" src="https://github.com/user-attachments/assets/715e0704-bec2-488f-b064-ac45a782b34a" />

staskus commented on WooCommerce/Classes/POS/Utils/POSFontStyle.swift:
```diff
@@ -34,8 +36,12 @@ enum POSFontStyle {
             Font.system(size: scaledValue(16, maximumContentSizeCategory: maximumContentSizeCategory), weight: .semibold)
-        case .posButtonSymbol:
-            Font.system(size: scaledValue(32, maximumContentSizeCategory: maximumContentSizeCategory), weight: .medium)
+        case .posButtonSymbolSmall:
```
We should set the font size here not the pixel size of the parent container for the icon. That's why the icons appear too large.

<img width="120" alt="image" src="https://github.com/user-attachments/assets/2def563b-1546-490b-8256-9155e8f26c90" />

iamgabrielma replied:
Ah! Sorry for that! 🤦🤦 I'm just confirming that sizes are consistent in Slack and will give it a new pass with the inner sizes rather than the container ones.

staskus commented:
> A bunch of icons currently used did not fit one of the 3 sizes provided, so I've used the closer possible. 

In case we set the correct values and we still see large differences between the designs and reality, let's answer these questions with a designer before making a PR. As a code reviewer, I cannot answer those questions as well. I only see a mismatch the same way you do.

iamgabrielma committed:
Update posButtonSymbol sizes

Mistakenly used the size proposed design for the outer container, rather than the font size itself.

iamgabrielma committed:
update item row to medium size

iamgabrielma committed:
update item placeholder icon to large

iamgabrielma committed:
update header banner collapsed icon to large

iamgabrielma committed:
revert PointOfSaleUnsupportedWidthView to original icon size

iamgabrielma committed:
revert 9776c479b4ea39674f51cb4bbc3d2e49c04026f0

We do not have explicit sizes for “extra large” icons which currently are displayed as 48/64px

iamgabrielma commented on WooCommerce/Classes/POS/Presentation/CardReaderConnection/UI States/PointOfSaleUnsupportedWidthView.swift:
```diff
@@ -10,7 +10,7 @@ struct PointOfSaleUnsupportedWidthView: View {
                 Image(systemName: "ipad.landscape.badge.exclamationmark")
                     .symbolRenderingMode(.palette)
-                    .foregroundStyle(Color(.wooCommerceAmber(.shade60)), Color.posPrimaryText)
+                    .foregroundStyle(Color.posAlert, Color.posAlert)
```
I opted to use the same colour here for consistency: 

<img width="911" alt="20250212_width_not_supported" src="https://github.com/user-attachments/assets/77dc1775-d573-4b0a-af90-6216340939a1" />

iamgabrielma replied:
Actually there are changes on a [newest PR](https://github.com/woocommerce/woocommerce-ios/pull/15119/files) that uses `.onSurface` and looks better, I'll adapt this one before merging:

![simulator_screenshot_D9766B86-6F85-42C8-9544-82638CFB6C23](https://github.com/user-attachments/assets/04ec8279-c224-4945-9d5e-a793d0ba34b9)

iamgabrielma commented:
This is ready for review again @staskus , apologies for the failed first attempt. I've updated the issue content above with a table with the changes, the component used, and the size used within that component just for easiness.

iamgabrielma committed:
update font weight from regular to semibold

iamgabrielma committed:
add preview

staskus reviewed (approved):
Thanks! Everything looks good now! 🚀

iamgabrielma committed:
Merge branch 'trunk' into task/15113-design-system-consolidate-icon-style

# Conflicts:
#	WooCommerce/Classes/POS/Presentation/CardReaderConnection/UI States/PointOfSaleUnsupportedWidthView.swift
#	WooCommerce/Classes/POS/Presentation/ItemListView.swift
#	WooCommerce/Classes/POS/Presentation/ItemRowView.swift
#	WooCommerce/Classes/POS/Utils/POSFontStyle.swift

iamgabrielma committed:
fix merge conflict
